### PR TITLE
Chore/service-name, conditionals

### PR DESCRIPTION
**Pull request by jenbutongit**
**changes made**
- fix conditionals 
- add form name to schema

### CODE REVIEW

File: page.js [hunk 1]
```diff
@@ -145,16 +146,33 @@ class Page {
...
+    }
+  }
+
+  setLangFromRequest (request) {
+    this.lang = request.query.lang || request.yar.get('lang') || 'en'
+    if (this.lang !== request.yar.get('lang')) {
+      request.i18n.setLocale(this.lang)
+      request.yar.set('lang', this.lang)
+    }
+  }
```
Comment (superafroman) [inline page.js:156]: Is `this` unique per request or shared?

Comment (jenbutongit) [reply page.js:156]: a `Page` is made for every page defined in the JSON.. not sure if it's unique per req though..? 🤔

Comment (superafroman) [reply page.js:156]: My guess is it's not, so we shouldn't store request-specific data on it - can you replace `this.lang` with `request.yar.get('lang')`, or `formData.lang` or ..?

Comment (jenbutongit) [reply page.js:156]: oki, I can revert back -- this is what I was doing before. I'll just pass in lang to `localisedString` instead

File: page.js [hunk 2]
```diff
@@ -145,16 +146,21 @@ class Page {
     return this.validate(newState, this.stateSchema)
   }
 
+  langFromRequest (request) {
+    let lang = request.query.lang || request.yar.get('lang') || 'en'
+    if (this.lang !== request.yar.get('lang')) {
```
Comment (superafroman) [inline page.js]: always `true` as `this.lang` is never set

**Review by superafroman — APPROVED** [review]

**Pull request merged**